src/main/java/com/google/sps/servlets/GetUserData.java

```diff
@@ -0,0 +1,119 @@
+    try {
+      authInstance = FirebaseAuth.getInstance(FirebaseAppManager.getApp());
+    } catch (IOException e) {
+      throw new ServletException(e);
+    }
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    System.setProperty(
+        DatastoreServiceConfig.DATASTORE_EMPTY_LIST_SUPPORT, Boolean.TRUE.toString());
+
+    try {
+      // Find user ID
+      String idToken = request.getParameter("idToken");
+      FirebaseToken decodedToken = authInstance.verifyIdToken(idToken);
+      String userID = decodedToken.getUid();
+      UserRecord userRecord = authInstance.getUser(userID);
+      String userEmail = userRecord.getEmail();
+
+      PreparedQuery queryUser =
+          datastore.prepare(
+              new Query("User")
+                  .setFilter(new FilterPredicate("userEmail", FilterOperator.EQUAL, userEmail)));
+
+      Entity userEntity;
+      // If the student user entity doesnt exist yet, create one
+      if (queryUser.countEntities() == 0) {
+        userEntity = new Entity("User");
+        userEntity.setProperty("userEmail", userEmail);
+        userEntity.setProperty("registeredClasses", Collections.emptyList());
+        userEntity.setProperty("ownedClasses", Collections.emptyList());
+        userEntity.setProperty("taClasses", Collections.emptyList());
+
+        datastore.put(userEntity);
+      } else {
+        userEntity = queryUser.asSingleEntity();
+      }
+
+      List<Key> registeredClassesList = (List<Key>) userEntity.getProperty("registeredClasses");
+      List<Key> ownedClassesList = (List<Key>) userEntity.getProperty("ownedClasses");
+      List<Key> taClassesList = (List<Key>) userEntity.getProperty("taClasses");
+
+      List<UserData> registeredClasses = new ArrayList<UserData>();
+      for (Key classKey : registeredClassesList) {
+        String classCode = KeyFactory.keyToString(classKey);
+        String className = (String) datastore.get(classKey).getProperty("name");
+
+        registeredClasses.add(new UserData(classCode, className));
+      }
+
+      List<UserData> ownedClasses = new ArrayList<UserData>();
+      for (Key classKey : ownedClassesList) {
+        String classCode = KeyFactory.keyToString(classKey);
+        String className = (String) datastore.get(classKey).getProperty("name");
+
+        ownedClasses.add(new UserData(classCode, className));
+      }
+
+      List<UserData> taClasses = new ArrayList<UserData>();
+      for (Key classKey : taClassesList) {
+        String classCode = KeyFactory.keyToString(classKey);
+        String className = (String) datastore.get(classKey).getProperty("name");
+
+        taClasses.add(new UserData(classCode, className));
+      }
+
+      response.setContentType("application/json;");
+      response.addHeader("registeredClasses", new Gson().toJson(registeredClasses));
```

Comment:
This data should not go in the headers of the response. They should be in the body:
`{"registeredClasses": "", "ownedClasses": "", ...}`